src/boot/efi/boot.c

```diff
@@ -2291,18 +2291,16 @@ static EFI_STATUS image_start(
 }
 
 static EFI_STATUS reboot_into_firmware(VOID) {
-        _cleanup_freepool_ CHAR8 *b = NULL;
-        UINTN size;
-        UINT64 osind;
+        UINT64 old, new;
         EFI_STATUS err;
 
-        osind = EFI_OS_INDICATIONS_BOOT_TO_FW_UI;
+        new = EFI_OS_INDICATIONS_BOOT_TO_FW_UI;
 
-        err = efivar_get_raw(EFI_GLOBAL_GUID, L"OsIndications", &b, &size);
+        err = efivar_get_uint64(EFI_GLOBAL_GUID, L"OsIndications", &old);
         if (!EFI_ERROR(err))
-                osind |= (UINT64)*b;
+                new |= old;
 
-        err = efivar_set_raw(EFI_GLOBAL_GUID, L"OsIndications", &osind, sizeof(UINT64), TRUE);
+        err = efivar_set_uint64(EFI_GLOBAL_GUID, L"OsIndications", new, TRUE);
```

Comment:
Ahhh, ok, that means some commits here are totally wrong since I replaced it with binary. Is there any explanation for why strings are used?

(I switched to binary because I couldn't find a string => 64-bit uint function in gnu-efi)

Reply:
binary sucks because of endianess issues. by using strings everything is clear. a lesson learned from udev too...

Reply:
I'll modify this to keep the original functions but rename them so it indicates these are formatted as strings and then add new functions for the binary variants for reading/writing UEFI stuff.